### ISSUE TITLE
Incorrect documentation?

### ISSUE THREAD

**Issue by oboxodo** [norman/friendly_id#649]
Hi there.

In the history modeule's documentation you use [this example](https://github.com/norman/friendly_id/blob/1d1479a8de25edf515ba618512e9e36dd6081f34/lib/friendly_id/history.rb#L49):

```ruby
    if request.path != post_path(@post)
      return redirect_to @post, :status => :moved_permanently
    end
```

My problem is that at least in the version of rails I'm using (4.1.6) request.path never includes the root forward slash.

```
(byebug) pp request.path
"cases/wille-v-southwestern-bell-telephone-co"
(byebug) pp case_path(@case, format: params[:format])
"/cases/wille-v-southwestern-bell-telephone-co"
```

I can solve this by appending the missing forward slash, but I wonder if this worked differently before, or if this is inconsistent behavior somehow.

What has been your experience?

**Comment by norman** [comment]
It definitely used to work, but it's actually been ages since I've used that in an application.

**Comment by oboxodo** [comment]
Aparently request.path DOES include the forward slash in test environment, but it doesn't in other environments.

request is an instance of ActionController::TestRequest in test, but otherwise it's an instance of ActionDispatch::Request. That might have something to do with it. Using your documentation example works on tests, but breaks in the wild :S.

Doesn't seem to be a problem directly related to FriendlyId, but maybe it's better to mention something in the docs about this.

**Comment by norman** [comment]
Ouch. I guess if that's the case, a better recommendation would be:

```ruby
if params[:id] != @post.friendly_id
  return redirect_to @post, :status => :moved_permanently
end

**Comment by oboxodo** [comment]
FTR for now I worked around it adding a private normalized_request_path method in ApplicationController and using it:

```ruby
  # For some reason (at least in Rails 4.1.6 + Rspec 2.14)
  # request#path only includes the initial forward slash in
  # test environment.
  def normalized_request_path
    request.path.start_with?('/') ? request.path : "/#{request.path}"
  end
```

**Comment by oboxodo** [comment]
Hey @norman, FYI this was a [bug in Rails 4.1.6](https://github.com/rails/rails/pull/17233). Fixed in Rails 4.1.8.

I upgraded my project to Rails 4.1.10 and now it works as expected. In any case, it might be a good idea to add a note to the suggested code mentioning that bug. Your call.

**Comment by norman** [comment]
Awesome - thanks for the followup, @oboxodo.
